packages/kbn-unified-data-table/src/components/data_table.tsx

```diff
@@ -828,6 +828,12 @@ export const UnifiedDataTable = ({
 
   const sorting = useMemo(() => {
     if (isSortEnabled) {
+      // in ES|QL mode, sorting is disabled when in Document view
+      // ideally we want the @timestamp column to be sortable server side
+      // but it needs discussion before moving forward like this
+      if (isPlainRecord && !columns.length) {
```

Comment:
Are we okay with allowing sorting when `@timestamp` is selected as a column? If not, we could set `isSortable` as false here https://github.com/elastic/kibana/blob/d296dc7c6739a3197e97bbfdc41a8273be719750/packages/kbn-unified-data-table/src/components/data_table_columns.tsx#L170-L173 or https://github.com/elastic/kibana/blob/d296dc7c6739a3197e97bbfdc41a8273be719750/packages/kbn-unified-data-table/src/components/data_table_columns.tsx#L221

<img width="1235" alt="Screenshot 2024-07-08 at 11 00 06" src="https://github.com/elastic/kibana/assets/1415710/37e8fa86-8888-4286-859d-74fbee2526d4">

Reply:
Thanx Jul! 

Yes at this scenario I am fine with allowing the client side sorting. I want to discuss with the team (the Discovery team) if it makes sense the @timestamp field and only to work server side (update the query with sort @timestamp) but it needs some discussion.